starfish/image/_registration/affine_transform.py

```diff
@@ -0,0 +1,123 @@
+from copy import deepcopy
+from itertools import product
+from typing import Optional, Union
+
+import numpy as np
+from skimage.transform import AffineTransform as AT
+from skimage.transform import warp
+
+from starfish.imagestack.imagestack import ImageStack
+from starfish.types import Axes
+from starfish.util import click
+from ._base import RegistrationAlgorithmBase
+
+
+class AffineTransform(RegistrationAlgorithmBase):
+    """
+    Performs an affine transform on the specified indices of an ImageStack
+
+    """
+    def __init__(
+        self, scale: Union[list, tuple]=None, rotation: float=None,
+        shear: float=None, translation: Union[list, tuple]=None,
+        rounds=None, channels=None, zplanes=None
+    ) -> None:
+
+        """Performs an affine transform on the specified indices of an ImageStack
+        Wraps skimage's AffineTransform
+        http://scikit-image.org/docs/dev/api/skimage.transform.html
+
+        Parameters
+        ----------
+        scale : list, tuple
+            images are registered to within 1 / upsample_factor of a pixel
+        rotation : float
+            Rotation angle in radians, counter-clockwise direction is positive
+        shear : float
+            Shear angle in radians, counter-clockwise direction is positive
+        translation : list, tuple
+            Number of pixels to translate the image (t_x, t_y)
+        rounds : np.ndarray
+            list of rounds to apply the transformation to. If omitted, transform
+            will be applied to all rounds.
+        channels : np.ndarray
+            list of channels to apply the transformation to. If omitted, transform
+            will be applied to all channels.
+        zplanes : np.ndarray
+            list of zplanes to apply the transformation to. If omitted, transform
+            will be applied to all zplanes.
+        """
+        self.scale = scale
+        self.rotation = rotation
+        self.shear = shear
+        self.translation = translation
+        self.rounds = rounds
+        self.channels = channels
+        self.zplanes = zplanes
+
+    def run(self, image: ImageStack, in_place: bool=False) -> Optional[ImageStack]:
+        """Register an ImageStack against a reference image.
+
+        Parameters
+        ----------
+        image : ImageStack
+            The stack to be registered
+        in_place : bool
+            If false, return a new registered stack. Else, register in-place (default False)
+
+        Returns
+        -------
+
+
+        """
+
+        if not in_place:
+            image = deepcopy(image)
+
+        # Get a list of all indices for an axes without provided indices
+        if self.rounds is None:
```

Comment:
We can't guarantee that channels, rounds, and zplanes are contiguous and zero-indexed. To account for this, use: 
```python3
if self.rounds is None: 
    self.rounds = image.coords[Axes.ROUND]
...
```